### PR TITLE
SAML: Allow disabling of SAML signups

### PR DESCRIPTION
**Pull request by mmandrus**
**What this PR does / why we need it**:
Allows system administrators to set the `[auth.saml]("allow_sign_ups")` option to `false` in their configuration. This decouples authentication with account creation, and allows SAML sign-ins only by users who have an account. This is needed to ensure orgs with user limits do not accidentally hit their cap due to automatic account creation.

Fixes https://github.com/grafana/grafana-enterprise/issues/3083

**Special notes for your reviewer**:
- This is my first time updating configuration & documentation, please confirm I updated everything that needs it
- Coincides with Enterprise PR https://github.com/grafana/grafana-enterprise/pull/3125

### CODE REVIEW

File: docs/sources/enterprise/saml.md [hunk 1]
```diff
@@ -142,6 +143,10 @@ For Grafana to map the user information, it looks at the individual attributes w
 
 Grafana provides configuration options that let you modify which keys to look at for these values. The data we need to create the user in Grafana is Name, Login handle, and email.
 
+### Allow new user signups
+
+By default, any new Grafana user who authenticates with SAML will have an account created for them automatically. To decouple authentication and account creation and ensure only users with existing accounts can log in with SAML, set the `allow_sign_up` option to false.
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/saml.md:148]:
```suggestion
By default, new Grafana users using SAML authentication will have an account created for them automatically. To decouple authentication and account creation and ensure only users with existing accounts can log in with SAML, set the `allow_sign_up` option to false.
```

File: docs/sources/enterprise/saml.md [hunk 2]
```diff
@@ -46,6 +46,7 @@ The table below describes all SAML configuration options. Continue reading below
 | ---------------------------------------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
 | `enabled`                                                  | No       | Whether SAML authentication is allowed                                                                                                                                                              | `false`       |
 | `single_logout`                                            | No       | Whether SAML Single Logout enabled                                                                                                                                                                  | `false`       |
+| `allow_sign_up`                                            | No       | Whether to allow new Grafana users to be created through SAML login. If set to `false`, only pre-existing Grafana users will be able to login with SAML.                                            | `true`        |
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/saml.md:49]:
```suggestion
| `allow_sign_up`                                            | No       | Whether to allow new Grafana user creation through SAML login. If set to `false`, then only existing Grafana users can log in with SAML.                                            | `true`        |
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Review by achatterjee-grafana — APPROVED** [review]

**Review by jalevin — APPROVED** [review]
LGTM

**Pull request merged**